compiler/src/main/scala/org/scalajs/core/compiler/JSGlobalAddons.scala

```diff
@@ -69,6 +69,20 @@ trait JSGlobalAddons extends JSDefinitions
       val destination: ExportDestination
     }
 
+    sealed abstract class JSName {
+      def displayName: String
+    }
+
+    object JSName {
+      final case class Literal(name: String) extends JSName {
```

Comment:
These non-static `final case class`es cause compilation warnings. It's a petty thing that scalac likes to do. I don't like it but they should non-final.